src/blob.rs

```diff
@@ -1,12 +1,55 @@
-use std::path::Path;
+use futures::{Async, Poll};
+use std::{
+    io,
+    marker::PhantomData,
+    path::{Path, PathBuf},
+};
+use tokio::{
+    fs::file::{File, OpenFuture, OpenOptions},
+    prelude::Future,
+};
 
 use crate::record::Record;
 
 /// A `Blob` struct for performing of database,
 #[derive(Debug, Default)]
-pub struct Blob<T> {
+pub struct Blob<K>
+where
+    K: Send,
+{
     header: Header,
-    records: Vec<T>, // @TODO needs verification, created to yield generic T up
+    records: Vec<K>, // @TODO needs verification, created to yield generic T up
```

Comment:
What kind of data this field is supposed to store? All keys? If so, why is it needed at all?